### PR TITLE
Updates as a result of changes to Proposed NB API

### PR DESCRIPTION
**Pull request by DonJayamanne**
For https://github.com/microsoft/vscode-jupyter/issues/5491

### CODE REVIEW

**Review by joyceerhl — APPROVED** [review]
Out of curiosity why do we have
1. types/vscode-proposed/index.d.ts
2. types/vscode.proposed.d.ts
3. typings/vscode-proposed/index.d.ts?

I see all three being updated in this PR.

**Comment by DonJayamanne** [comment]
> Out of curiosity why do we have

Good question. I have no idea. I think this is stuff coming from Python extension.

File: types/vscode-proposed/index.d.ts [hunk 1]
```diff
@@ -173,18 +173,46 @@ export interface NotebookDocument {
     readonly uri: Uri;
     readonly version: number;
 
+    /** @deprecated Use `uri` instead */
```
Comment (IanMatthewHuff) [inline types/vscode-proposed/index.d.ts:176]: Why do we seem to have three proposed API files? I only updated types/vscode.proposed.d.ts in my last API change.

**Review by IanMatthewHuff — APPROVED** [review]

File: types/vscode-proposed/index.d.ts (hunk 1 above)
Comment (DonJayamanne) [reply types/vscode-proposed/index.d.ts:176]: I'll remove the others in the separate PR.

**Pull request merged**